src/include/sof/samples/audio/smart_amp_test.h

```diff
@@ -81,10 +87,31 @@ typedef int(*smart_amp_proc)(struct comp_dev *dev,
  */
 
 struct sof_smart_amp_config {
+#if CONFIG_IPC_MAJOR_4
+	struct ipc4_base_module_cfg base;
+	uint16_t num_source_pin_fmt;
+	uint16_t num_sink_pin_fmt;
+	uint8_t reserved[8];
+	/* length of data after sink_pin_fmt */
+	uint32_t priv_param_len;
+	struct ipc4_input_pin_format source_pin_fmt[2];
```

Comment:
For a widget, sink pin is the input pin of the widget, and source pin is the output pin of the widget. I use the terms `sink` and `source` because they are already used in widget queue allocation in the kernel, and we have num_sink_pins (input) and num_source_pins (output) topology token. 

So the structure here for IPC4 should be:
```
uint16_t num_sink_pin_fmt; /* number of input pin formats */
uint16_t num_source_pin_fmt; /* number of output pin formats */
uint8_t reserved[8];
/* length of data after sink_pin_fmt */
uint32_t priv_param_len;
struct ipc4_input_pin_format sink_pin_fmt[2];
struct ipc4_output_pin_format source_pin_fmt;
```
of course, the firmware doesn't need to be the same as linux, you can use num_input_pin_fmt/num_output_pin_fmt

Reply:
It is a common usage that source is input and sink is output in fw, so I want to align with it

Reply:
it is common use sink for output in fw, do we need to use sink_pin for input ?